### PR TITLE
Add receivingChainID request param support to transactions endpoint

### LINKED ISSUE #1931: Add receivingChainID request param support to transactions endpoint
### Description

Allow user to filter transactions based on `receivingChainID`.

### Acceptance Criteria

- Add support for `receivingChainID` to the `GET /transactions` endpoint
- Update swagger
- Add integration tests
- Add unit tests
- Fix unit tests for transaction fee estimates.

### PR DESCRIPTION
**Pull request by vardan10**
### What was the problem?

This PR resolves #1931 

### How was it solved?
- [x] Add support for receivingChainID to the GET /transactions endpoint
- [x] Update swagger
- [x] Add integration tests
- [x] Add unit tests
- [x] Fix unit tests for transaction fee estimates.

### How was it tested?
Local

### CODE REVIEW

**Review by sameersubudhi — APPROVED** [review]

File: services/gateway/apis/http-version3/methods/transactions.js [hunk 1]
```diff
@@ -48,6 +48,7 @@ module.exports = {
 			max: 41,
 			pattern: regex.ADDRESS_LISK32,
 		},
+		receivingChainID: { optional: true, type: 'string', min: 1, pattern: regex.CHAIN_ID },
```
Comment (priojeetpriyom) [inline services/gateway/apis/http-version3/methods/transactions.js:51]: Already covered by regex.

```suggestion
		receivingChainID: { optional: true, type: 'string', pattern: regex.CHAIN_ID },
```

File: tests/integration/api_v3/http/transactions.test.js [hunk 2]
```diff
@@ -41,32 +42,56 @@ const baseAddress = config.SERVICE_ENDPOINT;
...
-				const response = await api.get(`${endpoint}?limit=1&moduleCommand=token:transfer`);
-				[refTransaction] = response.data;
-
-				if (refTransaction) {
-					success = true;
-				}
-			} catch (error) {
-				console.error(`Error fetching transactions. Retries left: ${retries}`);
-				retries--;
+		jest.setTimeout(Number.MAX_SAFE_INTEGER);
```
Comment (priojeetpriyom) [inline tests/integration/api_v3/http/transactions.test.js:77]: Better to set timeout at top of file for consistency.

File: tests/integration/api_v3/http/transactions.test.js [hunk 3]
```diff
@@ -346,6 +371,39 @@ describe('Transactions API', () => {
...
+	describe('Retrieve transaction list by receivingChainID', () => {
+		it('should return transactions when called with receivingChainID', async () => {
+			if (refTransaction.params.receivingChainID) {
+				const response = await api.get(
+					`${endpoint}?receivingChainID=${refTransaction.params.receivingChainID}`,
+				);
+				expect(response).toMap(goodRequestSchema);
+				expect(response.data).toBeInstanceOf(Array);
+				expect(response.data.length).toBeGreaterThanOrEqual(1);
+				expect(response.data.length).toBeLessThanOrEqual(10);
```
Comment (priojeetpriyom) [inline tests/integration/api_v3/http/transactions.test.js:383]: Isn't default limit 100? If so, then we should either update this check to 100 or add a limit param to the api call.

**Review by priojeetpriyom — APPROVED** [review]

File: tests/integration/api_v3/http/transactions.test.js (hunk 3 above)
Comment (sameersubudhi) [reply tests/integration/api_v3/http/transactions.test.js:383]: Default limit is always `10`.

**Review by priojeetpriyom — APPROVED** [review]

**Pull request merged**